e2etests/tests/validate.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.universe.tf/e2etest/pkg/frr"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	"go.universe.tf/e2etest/pkg/ipfamily"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func ValidateFRRNotPeeredWithNodes(nodes []corev1.Node, c *frrcontainer.FRR, ipFamily ipfamily.Family) {
+	for _, node := range nodes {
+		ginkgo.By(fmt.Sprintf("checking node %s is not peered with the frr instance %s", node.Name, c.Name))
+		Eventually(func() error {
+			neighbors, err := frr.NeighborsInfo(c)
+			framework.ExpectNoError(err)
+			err = frr.NeighborsMatchNodes([]corev1.Node{node}, neighbors, ipFamily, c.RouterConfig.VRF)
```

Comment:
Because if we do that, since we check the error, this will pass even if the session is not established  with one single node, whereas here we want to ensure the session is not established with all the nodes.

Reply:
ohh right. thanks